app/presenters/licences/view-licence.presenter.js

```diff
@@ -156,6 +147,75 @@ function _generateWarningMessage (ends) {
   return `This licence expired on ${formatLongDate(date)}`
 }
 
+function _parseAbstractionsAndSourceOfSupply (permitLicence) {
+  if (!permitLicence ||
+    permitLicence?.purposes === undefined ||
+    permitLicence.purposes.length === 0 ||
+    permitLicence.purposes[0]?.purposePoints === undefined ||
+    permitLicence.purposes[0]?.purposePoints.length === 0
+  ) {
+    return {
+      abstractionPoints: null,
+      abstractionPointsCaption: null,
+      abstractionPointLinkText: null,
+      sourceOfSupply: null
+    }
+  }
+
+  const abstractionPoints = []
+
+  if (permitLicence.purposes.length > 0) {
+    permitLicence.purposes.forEach((purpose) => {
+      purpose.purposePoints.forEach((point) => {
+        const pointDetail = point.point_detail
+        if (pointDetail) {
+          abstractionPoints.push(_generateAbstractionContent(pointDetail))
+        }
+      })
+    })
+  }
```

Comment:
Just a NIT. Do we need the check? If `permitLicences.purposes` is an empty array I think the `.forEach()` will simply never kick in with the same result. 🤷 

```suggestion
  permitLicence.purposes.forEach((purpose) => {
    purpose.purposePoints.forEach((point) => {
      const pointDetail = point.point_detail
      if (pointDetail) {
        abstractionPoints.push(_generateAbstractionContent(pointDetail))
      }
    })
  })
```

Reply:
I mostly added it in to accommodate the unit tests 😅  but I can update them to